### PR TITLE
git workflow for PR creation

### PR DESCRIPTION
**Pull request by tanvigour**
Follow entire formula generation and build workflow as soon as PR is created

### CODE REVIEW

File: .github/workflows/PR_workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          - os: macOS-latest
+            target_os: darwin
+    steps:
+      - name: Setup and build
+        if: github.event_name == 'pull_request'
+        run: |
+          brew help
+          DAPR_CLI_VERSION=$( curl -s -f "https://x-access-token:${GITHUB_TOKEN}@api.github.com/repos/dapr/cli/releases" | grep \"tag_name\" | awk 'NR==1{print $2}' |  sed -n 's/\"v\(.*\)\",/\1/p' )
+          echo "Found version ${DAPR_CLI_VERSION}"
+          echo "DAPR_CLI_VERSION=$DAPR_CLI_VERSION" >> $GITHUB_ENV
```
Comment (artursouza) [inline .github/workflows/PR_workflow.yaml:32]: No need to set env variables via GITHUB_ENV because they are not used outside this step.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: .github/workflows/PR_workflow.yaml (hunk 1 above)
Comment (tanvigour) [reply .github/workflows/PR_workflow.yaml:32]: they are used inside the generate scripts which are used in the next line

**Review by artursouza — APPROVED** [review]

File: .github/workflows/PR_workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+      - name: Get Dapr Cli's source code tarball's shasum.
+        env:
+          GITHUB_TOKEN: ${{ secrets.DAPR_BOT_TOKEN }}
+        if: env.NEW_VERSION == 'yes'
+        run: |
+          DAPR_CLI_SRC_SHASUM=$( curl -L -f "https://x-access-token:${GITHUB_TOKEN}@github.com/dapr/cli/archive/v${DAPR_CLI_VERSION}.tar.gz" 2> /dev/null | shasum -a 256 | cut -d\- -f1 | xargs -I {} echo {} )
+          echo "DAPR_CLI_SRC_SHASUM=$DAPR_CLI_SRC_SHASUM" >> $GITHUB_ENV
+      - name: Get Homebrew's tap for Dapr Cli.
+        if: env.NEW_VERSION == 'yes'
```
Comment (artursouza) [inline .github/workflows/PR_workflow.yaml:57]: Can this step also validate that all the versions in this repo can be installed?

**Review by artursouza — CHANGES_REQUESTED** [review]
Please, always do the validation, not only when it is a new version because a PR can happen at any time. The goal is to verify that every version in this repo can be installed.

Thanks.

**Pull request closed without merging**